OpenRA.Mods.Common/Widgets/Logic/Settings/GamePlaySettingsLogic.cs

```diff
@@ -0,0 +1,127 @@
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets.Logic
+{
+	public class GameplaySettingsLogic : ChromeLogic
+	{
+		[FluentReference]
+		const string AutoSaveIntervalOptions = "auto-save-interval.options";
+
+		[FluentReference]
+		const string AutoSaveIntervalDisabled = "auto-save-interval.disabled";
+
+		[FluentReference]
+		const string AutoSaveIntervalMinuteOptions = "auto-save-interval.minute-options";
+
+		[FluentReference]
+		const string AutoSaveMaxFileNumber = "auto-save-max-file-number";
+		readonly int[] autoSaveSeconds = { 0, 10, 30, 45, 60, 120, 180, 300, 600 };
+
+		readonly int[] atuoSaveFileNumbers = { 3, 5, 10, 20, 50, 100 };
+
+		[ObjectCreator.UseCtor]
+		public GameplaySettingsLogic(Action<string, string, Func<Widget, Func<bool>>, Func<Widget, Action>> registerPanel, string panelID, string label)
+		{
+			registerPanel(panelID, label, InitPanel, ResetPanel);
+		}
+
+		Func<bool> InitPanel(Widget panel)
+		{
+			var scrollPanel = panel.Get<ScrollPanelWidget>("SETTINGS_SCROLLPANEL");
+			SettingsUtils.AdjustSettingsScrollPanelLayout(scrollPanel);
+
+			// Setup dropdown for auto-save interval
+			var autoSaveIntervalDropDown = panel.Get<DropDownButtonWidget>("AUTO_SAVE_INTERVAL_DROP_DOWN");
+
+			autoSaveIntervalDropDown.OnClick = () =>
+				ShowAutoSaveIntervalDropdown(autoSaveIntervalDropDown, autoSaveSeconds);
+
+			autoSaveIntervalDropDown.GetText = () => GetMessageForAutoSaveInterval(Game.Settings.SinglePlayerSettings.AutoSaveInterval);
+
+			// Setup dropdown for auto-save nr.
+			var autoSaveNoDropDown = panel.Get<DropDownButtonWidget>("AUTO_SAVE_FILE_NUMBER_DROP_DOWN");
+
+			autoSaveNoDropDown.OnMouseDown = _ =>
+				ShowAutoSaveFileNumberDropdown(autoSaveNoDropDown, atuoSaveFileNumbers);
+
+			autoSaveNoDropDown.GetText = () => FluentProvider.GetMessage(AutoSaveMaxFileNumber, "saves", Game.Settings.SinglePlayerSettings.AutoSaveFileNumber);
+
+			autoSaveNoDropDown.IsDisabled = () => Game.Settings.SinglePlayerSettings.AutoSaveInterval <= 0;
+
+			return () => false;
+		}
+
+		Action ResetPanel(Widget panel)
+		{
+			return () => { };
+		}
+
+		void ShowAutoSaveIntervalDropdown(DropDownButtonWidget dropdown, IEnumerable<int> options)
+		{
+			var gsp = Game.Settings.SinglePlayerSettings;
+
+			ScrollItemWidget SetupItem(int o, ScrollItemWidget itemTemplate)
+			{
+				var item = ScrollItemWidget.Setup(itemTemplate,
+					() => gsp.AutoSaveInterval == o,
+					() =>
+					{
+						gsp.AutoSaveInterval = o;
+						Game.Settings.Save();
+					});
+
+				var deviceLabel = item.Get<LabelWidget>("LABEL");
+				deviceLabel.GetText = () => GetMessageForAutoSaveInterval(o);
+
+				return item;
+			}
+
+			dropdown.ShowDropDown("LABEL_DROPDOWN_TEMPLATE", 500, options, SetupItem);
+		}
+
+		void ShowAutoSaveFileNumberDropdown(DropDownButtonWidget dropdown, IEnumerable<int> options)
+		{
+			var gsp = Game.Settings.SinglePlayerSettings;
+
+			ScrollItemWidget SetupItem(int o, ScrollItemWidget itemTemplate)
```

Comment:
Can this setting/option be change while in game? What would happen if number of files first was on the max and after set to min. Then the game would pause to delete up to 600-10 files?

Reply:
yes